### ISSUE TITLE
Perfomance problems on large sites with user logins

### ISSUE THREAD

**Issue by nmalcolm** [mybb/mybb#349]
> Currently the username login is using <acronym title="">LOWER</acronym> in the SQL query. This means the index is not used on some Database backends.
> 
> Forums such as ours (150k+ users) slow down when spammers try to login with spambots all the time.
> 
> We suggest you remove the LOWER from the SQL so the index is used and it becomes fast again.
> 
> Its used in a number of places in the code so we would rather this was fixed upstream.

Original report: http://dev.mybb.com/issues/2290
Reporter: zag zag
Status: New

**Comment by ATofighi** [comment]
@JN-Jones yes, https://github.com/mybb/mybb/blob/feature/inc/db_mysql.php#L1440-L1476.

**Comment by JN-Jones** [comment]
It's not about how they're named - it's about how they handle queries:

> By default, we use utf8_general_ci (case insensitive) so we can drop the LOWER() calls.

I'm not sure as the mysql docs seems to say something different: http://dev.mysql.com/doc/refman/5.0/en/case-sensitivity.html

**Comment by Destroy666x** [comment]
Also, the PR fixed only `LOWER` username queries, what about Emails for example? https://github.com/mybb/mybb/pull/1724/files#diff-7fe6e55397c77ab9a0f5d57bc4cbe5b9R5679

And this var can be cleaned now since you changed to input: https://github.com/mybb/mybb/blob/3e3b1933c08e89adc2cfaf02bb1c6c27ff761c04/admin/modules/tools/maillogs.php#L176

**Comment by Sama34** [comment]
So is the performance gain still present if we do need to check for the DB engine plus the encoding type?

**Comment by ATofighi** [comment]
I try to install MyBB with binary encoding but I see SQL Error...
![mybb installation wizard - table creation](https://cloud.githubusercontent.com/assets/3256261/5603592/f03db380-93a1-11e4-9a79-bedda6f24a40.jpg)

**Comment by Destroy666x** [comment]
@Sama34, checking the db and encoding is a matter of microsecond(s) in PHP (the switch you already have + an additional if for encoding). Using `LOWER()` on many MySQL rows takes much longer, especially if the database is large.

@ATofighi, there is an issue for this which is awaiting to be fixed too: https://github.com/mybb/mybb/issues/219

**Comment by Sama34** [comment]
How do you check the encoding? Is it '$config['database']['encoding']' with 'utf8' as value?

**Comment by Sama34** [comment]
@Destroy666x @mybb/developers

**Comment by Destroy666x** [comment]
@Sama34 , I think this should be decided first: https://github.com/mybb/mybb/issues/219

**Comment by JN-Jones** [comment]
Closing this as the issue here has been fixed. #219 needs to be discussed and that's set for 1.8.5

**Comment by Destroy666x** [comment]
It's not fully fix though even if we skip the binary collation, see 2 comments above:

> Also, the PR fixed only LOWER username queries, what about Emails for example? https://github.com/mybb/mybb/pull/1724/files#diff-7fe6e55397c77ab9a0f5d57bc4cbe5b9R5679
And this var can be cleaned now since you changed to input: https://github.com/mybb/mybb/blob/3e3b1933c08e89adc2cfaf02bb1c6c27ff761c04/admin/modules/tools/maillogs.php#L176

**Comment by JN-Jones** [comment]
Ah, missed that comment, sorry for that. Can you create a PR for those?

**Comment by Destroy666x** [comment]
I can fix the Email `LOWER` call(s) and remove the variable after I finish my 2 remaining assigments. Unless someone else wants to.

However, there are still lots of LOWER calls in inc/function_search.php which could be optimised too I guess, but that's not connected with logins and another issue should probably be made for 1.8.5.

**Comment by JN-Jones** [comment]
I'd guess there're a lot more places where we make case sensitive queries. But as 1.8.4 is near we should create a new issue.
